client/ping.go

```diff
@@ -26,5 +27,10 @@ func (cli *Client) Ping(ctx context.Context) (types.Ping, error) {
 		ping.Experimental = true
 	}
 
+	serverHeader := serverResp.header.Get("Server")
+	if strings.Contains(serverHeader, "()") {
+		ping.ServerOS = serverHeader[strings.Index(serverHeader, "(")+1 : strings.Index(serverHeader, ")")]
+	}
```

Comment:
This seems kind of hacky, I can see this failing unexpectedly.

I think it would be better to add a new header with the serverOS in a format that is easier to parse (or doesn't need to be parsed at all).